go/parquet/schema/converted_types.go

```diff
@@ -0,0 +1,191 @@
+		return "UNKNOWN"
+	default:
+		return format.ConvertedType(p).String()
+	}
+}
+
+// ToLogicalType returns the correct LogicalType for the given ConvertedType, using the decimal
+// metadata provided to define the precision/scale if necessary
+func (p ConvertedType) ToLogicalType(convertedDecimal DecimalMetadata) LogicalType {
+	switch p {
+	case ConvertedTypes.UTF8:
+		return StringLogicalType{}
+	case ConvertedTypes.Map, ConvertedTypes.MapKeyValue:
+		return MapLogicalType{}
+	case ConvertedTypes.List:
+		return ListLogicalType{}
+	case ConvertedTypes.Enum:
+		return EnumLogicalType{}
+	case ConvertedTypes.Decimal:
+		return NewDecimalLogicalType(convertedDecimal.Precision, convertedDecimal.Scale)
+	case ConvertedTypes.Date:
+		return DateLogicalType{}
+	case ConvertedTypes.TimeMillis:
+		return NewTimeLogicalType(true, TimeUnitMillis)
+	case ConvertedTypes.TimeMicros:
+		return NewTimeLogicalType(true, TimeUnitMicros)
+	case ConvertedTypes.TimestampMillis:
+		t := NewTimestampLogicalType(true, TimeUnitMillis)
+		t.(*TimestampLogicalType).fromConverted = true
+		return t
+	case ConvertedTypes.TimestampMicros:
+		t := NewTimestampLogicalType(true, TimeUnitMicros)
+		t.(*TimestampLogicalType).fromConverted = true
+		return t
+	case ConvertedTypes.Interval:
+		return IntervalLogicalType{}
+	case ConvertedTypes.Int8:
+		return NewIntLogicalType(8, true)
+	case ConvertedTypes.Int16:
+		return NewIntLogicalType(16, true)
+	case ConvertedTypes.Int32:
+		return NewIntLogicalType(32, true)
+	case ConvertedTypes.Int64:
+		return NewIntLogicalType(64, true)
+	case ConvertedTypes.Uint8:
+		return NewIntLogicalType(8, false)
+	case ConvertedTypes.Uint16:
+		return NewIntLogicalType(16, false)
+	case ConvertedTypes.Uint32:
+		return NewIntLogicalType(32, false)
+	case ConvertedTypes.Uint64:
+		return NewIntLogicalType(64, false)
+	case ConvertedTypes.JSON:
+		return JSONLogicalType{}
+	case ConvertedTypes.BSON:
+		return BSONLogicalType{}
+	case ConvertedTypes.None:
+		return NoLogicalType{}
+	case ConvertedTypes.NA:
+		fallthrough
+	default:
+		return UnknownLogicalType{}
+	}
+}
+
+// GetSortOrder defaults to the sort order based on the physical type if convert
+// is ConvertedTypes.None, otherwise determines the sort order by the converted type.
+func GetSortOrder(convert ConvertedType, primitive format.Type) SortOrder {
+	if convert == ConvertedTypes.None {
+		return DefaultSortOrder(primitive)
+	}
+	switch convert {
+	case ConvertedTypes.Int8,
+		ConvertedTypes.Int16,
+		ConvertedTypes.Int32,
+		ConvertedTypes.Int64,
+		ConvertedTypes.Date,
+		ConvertedTypes.TimeMicros,
+		ConvertedTypes.TimeMillis,
+		ConvertedTypes.TimestampMicros,
+		ConvertedTypes.TimestampMillis:
+		return SortSIGNED
+	case ConvertedTypes.Uint8,
+		ConvertedTypes.Uint16,
+		ConvertedTypes.Uint32,
+		ConvertedTypes.Uint64,
+		ConvertedTypes.Enum,
+		ConvertedTypes.UTF8,
+		ConvertedTypes.BSON,
+		ConvertedTypes.JSON:
+		return SortUNSIGNED
+	case ConvertedTypes.Decimal,
```

Comment:
this doesn't look right for decimal.  I believe it should be signed?  Or might vary based on underlying type.

Reply:
so i definitely copied this from the C++ implementation, i believe it's "sortUNKNOWN" because it can differ based on the underlying type, that said it means my current implementation of statistic sorting for decimals is incorrect now that i read the spec :) so i'll fix that in the rest of my code so it's correct before I put that part of the chunks up haha. 

That said, given that int32 and int64 as the underling types for the decimal are signed comparison, i think i'll shift this to being marked as SortSIGNED since the spec defines decimal as sort by signed comparison.